### DISCUSSION TITLE
New Blueprint: Display Value with Icon and Colors

### DISCUSSION THREAD

**Discussion by aderusha** [HASwitchPlate/HASPone#152] [Show and tell]
[![Display Value with Icon and Colors](https://my.home-assistant.io/badges/blueprint_import.svg)](https://my.home-assistant.io/redirect/blueprint_import/?blueprint_url=https%3A%2F%2Fgithub.com%2FHASwitchPlate%2FHASPone%2Fblob%2Fmain%2FHome_Assistant%2Fblueprints%2Fhasp_Display_Value_with_Icon_and_Colors.yaml)

![Preview](https://raw.githubusercontent.com/HASwitchPlate/HASPone/main/images/hasp_Display_Value_with_Icon_and_Colors.png)

A HASP button displays the current value of an entity (state or attribute) with a dynamic icon on the left and (optional) colors.  Up to 5 icons and color ranges are supported.

This is a generalization of "Display Weather Temperature with Icon and Colors", allowing the user to select any entity and to extract the value of the entity state or attribute.  Up to 5 ranges can be defined which will be used to determine the icon displayed and the color used for the icon (and optionally, the text).

**Comment by IIAIronWolf** [comment]
Awesome!
